src/basic/chattr-util.c

```diff
@@ -172,3 +172,21 @@ int read_attr_at(int dir_fd, const char *path, unsigned *ret) {
 
         return read_attr_fd(fd, ret);
 }
+
+int set_proj_id(int dir_fd, uint32_t proj_id) {
```

Comment:
dir_fd is a bit of a misnomer, isn't it? afair you can set this also on a regular file?